aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -342,6 +352,75 @@ module aptos_framework::account {
         rotation_capability
     }
 
+    public entry fun offer_signer_capability(account: &signer, signer_capability_sig_bytes: vector<u8>, account_scheme: u8, account_public_key_bytes: vector<u8>, recipient_address: address) acquires Account {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr) && exists_at(recipient_address), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+        assert!(account_scheme == 0 || account_scheme == 1, error::invalid_argument(EINVALID_SCHEME));
+
+        let account_resource = borrow_global_mut<Account>(addr);
+        let signer_capability_offer_proof_challenge = SignerCapabilityOfferProofChallenge {
+            sequence_number: account_resource.sequence_number,
+            recipient_address,
+        };
+
+        if (account_scheme == ED25519_SCHEME) {
```

Comment:
why do we need to distinguish multi-sig and single sig here?  My understanding this is signer authority delegation. the signer is the same for multi-sig and single sig. And the proof of og signer owner bytes would be the same as well.

Reply:
because we wanted to make sure that the public key matches the account's auth key, and the methods for creating `new_unvalidated_public_key_from_bytes` and `unvalidated_public_key_to_authentication_key` are different for ed25519 and multi-ed25519 key?

Reply:
I see. my confusion is why we need this guard. Could you help me understand the reasoning?
Previously, key rotation requires this proof as we don't want signer delegators to acquire the entire account in one txn.

Now, suppose a malicious signer delegator want to provide his signer to another one. I don't see the risk is higher if we allow this.